apps/language_models/src/pipelines/falcon_pipeline.py

```diff
@@ -74,33 +83,48 @@ def __init__(
         debug=False,
     ) -> None:
         super().__init__(model_name, hf_model_path, max_num_tokens)
+        print("hf_model_path: ", self.hf_model_path)
+
+        if "180b" in self.model_name and hf_auth_token == None:
+            raise ValueError(
+                """ HF auth token required for falcon-180b. Pass it using
+                --hf_auth_token flag. You can ask for the access to the model
+                here: https://huggingface.co/tiiuae/falcon-180B-chat."""
+            )
+        self.hf_auth_token = hf_auth_token
         self.max_padding_length = 100
         self.device = device
         self.precision = precision
         self.falcon_vmfb_path = falcon_vmfb_path
         self.falcon_mlir_path = falcon_mlir_path
         self.debug = debug
         self.tokenizer = self.get_tokenizer()
-        self.shark_model = self.compile()
         self.src_model = self.get_src_model()
+        self.shark_model = self.compile()
 
     def get_tokenizer(self):
         tokenizer = AutoTokenizer.from_pretrained(
-            self.hf_model_path, trust_remote_code=True
+            self.hf_model_path,
+            trust_remote_code=True,
+            token=self.hf_auth_token,
         )
```

Comment:
Can we get rid of this with a similar approach to what's being done for llama in `vicuna.py`?
This in turn should mean that we can perhaps get rid of `hf_auth_token`?

Reply:
I don't think the same can be done for Falcon now. Since the Falcon pipeline needs the original model during the inference as well. Also, the token is required just for the falcon-180b which won't be used once I add another patch for the 180b-gptq variant because that doesn't require any auth token.